### DISCUSSION TITLE
后续版本中保留 Pagination 组件目前所有 Event 事件

### DISCUSSION THREAD

**Discussion by hooray** [element-plus/element-plus#12843] [Polls]
文档中建议：

> 以上事件不推荐使用（但由于兼容的原因仍然支持，在以后的版本中将会被删除）；如果要监听 current-page 和 page-size 的改变，使用 v-model 双向绑定是个更好的选择。

虽然我可以通过 v-model 双向绑定，并监听 current-page 去触发列表查询接口实现分页效果，但这样会涉及到一个业务场景的问题：

假如我有筛选功能，当点击筛选按钮后，需要将 current-page 设为 1 并触发列表查询接口，这时候如果处于第一页（current-page 为 1）则并没有问题，如果不在第一页上，则会触发两次列表查询接口，一次是筛选按钮点击触发的，一次是监听 current-page 函数里触发的。

类似的场景同样存在于 page-size 上，当修改每页条目数后，业务场景下的需求通常是要重置到第一页，所以我会监听 page-size 的变化，将 current-page 设为 1 并触发列表查询接口，后面的问题则与上面一致。

[示例链接](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIEBjbGljaz1cInNlYXJjaFwiPuaooeaLn+aQnOe0ouaMiemSrjwvZWwtYnV0dG9uPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1wYWdpbmF0aW9uLWJsb2NrXCI+XG4gICAgPGVsLXBhZ2luYXRpb25cbiAgICAgIHYtbW9kZWw6Y3VycmVudC1wYWdlPVwiY3VycmVudFBhZ2VcIlxuICAgICAgdi1tb2RlbDpwYWdlLXNpemU9XCJwYWdlU2l6ZVwiXG4gICAgICA6cGFnZS1zaXplcz1cIlsxMDAsIDIwMCwgMzAwLCA0MDBdXCJcbiAgICAgIGxheW91dD1cInRvdGFsLCBzaXplcywgcHJldiwgcGFnZXIsIG5leHQsIGp1bXBlclwiXG4gICAgICA6dG90YWw9XCI0MDBcIlxuICAgIC8+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYsIHdhdGNoIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBjdXJyZW50UGFnZSA9IHJlZigxKVxuY29uc3QgcGFnZVNpemUgPSByZWYoMTAwKVxuXG53YXRjaCgoKSA9PiBjdXJyZW50UGFnZS52YWx1ZSwgKCkgPT4ge1xuICBnZXRMaXN0KClcbn0pXG4gIFxud2F0Y2goKCkgPT4gcGFnZVNpemUudmFsdWUsICgpID0+IHtcbiAgY3VycmVudFBhZ2UudmFsdWUgPSAxXG4gIGdldExpc3QoKVxufSlcblxuZnVuY3Rpb24gZ2V0TGlzdCgpIHtcbiAgY29uc29sZS5sb2coJ+aooeaLn+ivt+axguWIl+ihqOaOpeWPoycpXG59XG4gIFxuZnVuY3Rpb24gc2VhcmNoKCkge1xuICBjdXJyZW50UGFnZS52YWx1ZSA9IDFcbiAgZ2V0TGlzdCgpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

所以组件目前提供的 size-change 和 current-change 这两个 event 还是有必要存在的，我可以直接通过这两个 event 去触发接口请求，而不需要通过监听 current-page 的值去触发。

**Comment by mymx2** [comment]
虽然保留比较好，但也能从requestLoading状态或者叫paginationStop状态进行判断。
```
paginationStop = true
currentPage.value = 1
```

**Comment by qqzhouhaijun** [comment]
哈哈我也发现了.然后研究了半天找不到解决办法,

**Comment by xingyixiang** [comment]
我想通过这个pr解决这个问题
[#14828](https://github.com/element-plus/element-plus/pull/14828)
